sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConnectionStringProperties.java

```diff
@@ -0,0 +1,117 @@
+ *     <li>SharedAccessKeyName and SharedAccessKey, optional, used to authenticate the access to the EventHub.</li>
+ *     <li>SharedAccessSignature, optional, an alternative way to authenticate the access to the EventHub.</li>
+ *     <li>EntityPath, optional, the queue name or the topic name under the service namespace</li>
+ * </ul>
+ *
+ * <p>
+ * When you have a EventHub connection string, you can use {@link EventHubClientBuilder#connectionString(String)}
+ * to build a client. If you'd like to use a {@link TokenCredential} to access a Event Hub, you can use this utility
+ * class to take the fully qualified namespace and optionally the entity path (queue/topic name) from the connection
+ * string and then use {@link EventHubClientBuilder#credential(String, String, TokenCredential)}.
+ * </p>
+ *
+ * @see EventHubClientBuilder#connectionString(String)
+ */
+public final class EventHubConnectionStringProperties {
+    private final URI endpoint;
+    private final String entityPath;
+    private final String sharedAccessKeyName;
+    private final String sharedAccessKey;
+    private final String sharedAccessSignature;
+
+    private EventHubConnectionStringProperties(ConnectionStringProperties properties) {
+        this.endpoint = properties.getEndpoint();
+        this.entityPath = properties.getEntityPath();
+        this.sharedAccessKeyName = properties.getSharedAccessKeyName();
+        this.sharedAccessKey = properties.getSharedAccessKey();
+        this.sharedAccessSignature = properties.getSharedAccessSignature();
+    }
+    /**
+     * Parse a Event Hub connection string into an instance of this class.
+     *
+     * @param connectionString The connection string to be parsed.
+     *
+     * @return An instance of this class.
+     * @throws NullPointerException if {@code connectionString} is null.
+     * @throws IllegalArgumentException if the {@code connectionString} is empty or malformed.
+     */
+    public static EventHubConnectionStringProperties parse(String connectionString) {
+        return new EventHubConnectionStringProperties(new ConnectionStringProperties(connectionString));
+    }
+
+    /**
+     * Gets the "EntityPath" value of the connection string.
+     *
+     * @return The entity path, or {@code null} if the connection string doesn't have an "EntityPath".
+     */
+    public String getEntityPath() {
+        return entityPath;
+    }
+
+    /**
+     * Gets the "Endpoint" value of the connection string.
+     *
+     * @return The endpoint.
+     */
+    public String getEndpoint() {
+        return String.format("%s://%s", endpoint.getScheme(), endpoint.getHost());
+    }
+
+    /**
+     * Gets the fully qualified namespace, or hostname, from the connection string "Endpoint" section.
+     *
+     * @return The fully qualified namespace.
+     */
+    public String getFullyQualifiedNamespace() {
+        return this.endpoint.getHost();
```

Comment:
nit: consistent use of `this`. In the method above, you did not reference member variables with `this`. In general, unless there is a naming conflict, I would remove `this`.

Reply:
According to your comment, fixed in new version.